src/domain/data-decoder/entities/schemas/data-decoded.schema.ts

```diff
@@ -0,0 +1,31 @@
+import { Schema } from 'ajv';
+
+export const dataDecodedParameterSchema: Schema = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    param_type: { type: 'string' },
+    value: { anyOf: [{ type: 'string' }, { type: 'number' }] },
+    valueDecoded: {
+      anyOf: [
+        { type: 'object', nullable: true },
+        { type: 'array', nullable: true },
+      ],
+    },
+  },
+  required: ['name', 'value'],
+  additionalProperties: true,
+};
+
+export const dataDecodedSchema: Schema = {
```

Comment:
Why not constrain the schema with `JSONSchemaType`? It would give some better error hints if the schema doesn't match a type `T`.

Reply:
This is something I think we discussed when introduced AJV validation... for some reason `JSONSchemaType` is not fully compatible with nullable fields (TS compilation fails), or maybe I'm missing something in the documentation. You can try to change the definition to see if you find the way. We could create a new issue to investigate this further if you like, but I was already blocked by this for a while in the past.

Another solution, in the case of the data-decoder endpoint, due to its nature (payloads are pretty free I think), we could just skip validation for some fields in the payload as I suggested, wdyt?

Reply:
Please refer to [the previous comment](https://github.com/5afe/safe-client-gateway-nest/pull/132#discussion_r998412182)